### PR TITLE
Adds functionality to dice

### LINKED ISSUE #4707: Functioning dice, that you can roll.
## Description
We could use dice, that you can roll. 
Features:
- A range of numbered faces that a normal die can possibly land on (such as 1 to 6).
- Dice with special faces that they can land on. For example, the 4d d6 with its "Cube-Side: 1-1", "Cube-Side: 4-4", "Cube-Side: 8-6", and many other strange faces.
- The ability to roll the dice by either using it in-hand or throwing it.
- Messages that show in chat when a die is rolled and lands on a face.
- An overlay sprite on the die that updates depending on the face landed on (if applicable).
- The ability to rig dice by microwaving them.
- Special messages if you roll a 1 or a 20 on a d20 (twenty-sided die).

It should be noted that all dice items from /tg/ are in the game right now and have .prefabs, though they don't do anything yet.

/tg/ .dm code below.
image

### PR DESCRIPTION
**Pull request by NoooneyDude**
### Purpose
- Dice can now be rolled. Includes all of the features described in #4707 (fixes #4707).
- Moves dice files into sub-folders based on the die involved.

### CODE REVIEW

**Comment by Bod9001** [comment]
looks good

File: UnityProject/Assets/Scripts/Items/Dice/RollDie.cs [hunk 1]
```diff
@@ -0,0 +1,225 @@
...
+	}
+
+	/// <summary>
+	/// Clamps the given integer to within the given minimum and maximum values.
+	/// </summary>
+	/// <param name="given">The value to be clamped</param>
+	/// <param name="min">The minimum possible value</param>
+	/// <param name="max">The maximum possible value</param>
+	/// <returns>A clamped value between the minimum and maximum values.</returns>
+	protected int ClampInt(int given, int min, int max)
```
Comment (corp-0) [inline UnityProject/Assets/Scripts/Items/Dice/RollDie.cs]: ``Mathf.Clamp()``

File: UnityProject/Assets/Scripts/Items/Dice/RollDie.cs [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+
+/// <summary>
+/// An extendable MonoBehaviour component to handle generic die rolling.
+/// </summary>
+public class RollDie : MonoBehaviour, IExaminable, ICheckedInteractable<HandActivate>
+{
+	[Tooltip("The amount of sides this die has.")]
+	public int sides = 6;
+	[Tooltip("Whether the die can be rigged to give flawed results.")]
+	public bool riggable = true;
```
Comment (nuke-haus) [inline UnityProject/Assets/Scripts/Items/Dice/RollDie.cs:15]: Public properties should be capitalized, for future reference

**Comment by nuke-haus** [comment]
I made a few small tweaks to the code, mostly there were just small naming/formatting issues. Otherwise looks good to me.

File: UnityProject/Assets/Scripts/Items/Dice/RollDie.cs (hunk 2 above)
Comment (DooblyNoobly) [reply UnityProject/Assets/Scripts/Items/Dice/RollDie.cs:15]: These are fields not properties. So fields should be camelCase and properties should be PascalCase

**Comment by nuke-haus** [comment]
Sorry about that, i realize it's probably bad manners to modify someone's PR. in the future i'm gonna stick to just comments

**Pull request merged**